axelrod/strategies/titfortat.py

```diff
@@ -92,6 +93,41 @@ class TwoTitsForTat(Player):
     def strategy(opponent: Player) -> Action:
         return D if D in opponent.history[-2:] else C
 
+class DynamicTwoTitsForTat(Player):
+    """
+    A player starts by cooperating and then punishes its opponent's 
+    defectiions by opponent with a dynamic bias based off of the 
+    opponents ratio of cooperations to total moves (so their current 
+    probability of cooperating towards cooporating regardless of the 
+    move (aka: forgiveness)).
+    
+    Names:
+
+     - Dynamic Two Tits For Tat: Original name by Grant Garrett-Grossman.
+    """
+
+    name = 'Dynamic Two Tits For Tat'
+    classifier = {
+        'memory_depth': 2,  # Long memory, memory-2
+        'stochastic': True,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+    
+    @staticmethod
+    def strategy(opponent):
+        # First move
+        if len(opponent.history) == 0:
```

Comment:
FYI @FakeNameSE this gives some good background: https://stackoverflow.com/questions/53513/best-way-to-check-if-a-list-is-empty

(Apologies, as is was my suggestion, thanks to @meatballs for pointing it out!)